discovery-provider/src/tasks/index.py

```diff
@@ -951,6 +957,22 @@ def revert_blocks(self, db, revert_blocks_list):
                 logger.info(f"Reverting follow: {follow_to_revert}")
                 session.delete(follow_to_revert)
 
+            for subscription_to_revert in revert_subscription_entries:
+                previous_subscription_entry = (
+                    session.query(Subscription)
+                    .filter(
+                        Subscription.subscriber_id
+                        == subscription_to_revert.subscriber_id
+                    )
+                    .filter(Subscription.user_id == subscription_to_revert.user_id)
+                    .order_by(Subscription.blocknumber.desc())
+                    .first()
+                )
```

Comment:
this should have another filter on blocknumber < revert_block_number
See for loop below

Reply:
i think it was copied from follows which should have that too

Reply:
seems like there may be follows with no is_current then